pkgs/development/python-modules/mmcv/default.nix

```diff
@@ -0,0 +1,93 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, python3
+, pythonOlder
+, requests
+, cudaSupport ? false
+, cudatoolkit
+, torch-cuda-arch-list ? "Kepler"
+, torch
+, torchvision
+, torch-bin
+, torchvision-bin
+, opencv4
+, yapf
+, packaging
+, pillow
+, addict
+, ninja
+, which
+, onnx
+, onnxruntime
+, scipy
+, pyturbojpeg
+, tifffile
+, lmdb
+}:
+
+buildPythonPackage rec {
+  pname = "mmcv";
+  version = "1.7.0";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "open-mmlab";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-EVu6D6rTeebTKFCMNIbgQpvBS52TKk3vy2ReReJ9VQE=";
+  };
+
+  torch-deps = if cudaSupport then [ torch-bin torchvision-bin ] else [ torch torchvision ];
+  nativeBuildInputs = [ ninja which ]
+    ++ lib.lists.optionals cudaSupport [ cudatoolkit.lib cudatoolkit.out cudatoolkit.cc ] ++ torch-deps;
+
+  postPatch = ''
+    substituteInPlace setup.py --replace "cpu_use = 4" "cpu_use = $NIX_BUILD_CORES"
+  '';
+
+  # reason for not using pytestCheckHook is similiar to what
+  # is already mentioned in python-module/typed-ast
+  # test_cnn test_ops really requires gpus to be useful.
+  # some of the tests take exceedingly long time.
+  # the rest of the tests are disabled due to sandbox env.
+  checkPhase = ''
+    pytest --ignore=tests/test_cnn \
+           --ignore=tests/test_ops \
+           --ignore=tests/test_fileclient.py \
+           --ignore=tests/test_load_model_zoo.py \
+           --ignore=tests/test_runner/test_checkpoint.py \
+           --ignore=tests/test_video/test_processing.py \
+           --ignore=tests/test_utils/test_hub.py \
+           --ignore=tests/test_video/test_reader.py
+  '';
+
+  checkInputs = [ lmdb onnx onnxruntime scipy pyturbojpeg tifffile ];
+
+  propagatedBuildInputs = [
+    opencv4
+    yapf
+    packaging
+    pillow
+    addict
+  ] ++ torch-deps;
+
+  pythonImportsCheck = [
+    "mmcv"
+  ];
+
+  MMCV_WITH_OPS = 1;
+  FORCE_CUDA = cudaSupport;
+  TORCH_CUDA_ARCH_LIST = torch-cuda-arch-list;
+  LDFLAGS = lib.optionalString cudaSupport "-L${cudatoolkit.lib}/lib";
+
+  meta = with lib; {
+    description = "MMCV is a foundational library for computer vision research";
```

Comment:
as per contribution guidelines, description should not start or contain name of the package

Reply:
fixed